solana/programs/example-native-token-transfers/src/instructions/redeem.rs

```diff
@@ -96,7 +98,17 @@ pub fn redeem(ctx: Context<Redeem>, _args: RedeemArgs) -> Result<()> {
     let message: NttManagerMessage<NativeTokenTransfer> =
         accs.transceiver_message.message.ntt_manager_payload.clone();
 
-    let amount = message.payload.amount.untrim(accs.mint.decimals);
+    // Calculate the scaled amount based on the appropriate decimal encoding for the token.
+    // Return an error if the resulting amount overflows.
+    // Ideally this state should never be reached: the sender should avoid sending invalid
+    // amounts when they would cause an error on the receiver.
+    let amount = match message.payload.amount.untrim(accs.mint.decimals) {
+        Ok(amount) => amount,
+        Err(ScalingError::OverflowExponent) => return Err(error!(NTTError::OverflowExponent)),
+        Err(ScalingError::OverflowScaledAmount) => {
+            return Err(error!(NTTError::OverflowScaledAMount))
+        }
+    };
```

Comment:
instead of duplicating these error mappings everywhere, we could implement a `impl From<ScalingError> for NTTerror` in the `error.rs` file and here (and everywhere else) just do
```suggestion
    let amount = match message.payload.amount.untrim(accs.mint.decimals).map_err(|e| e.into())?;
```